nav-app/src/app/techniques-search/techniques-search.component.ts

```diff
@@ -37,71 +39,140 @@ export class TechniquesSearchComponent implements OnInit {
         }
     ]
 
+    private previousQuery: string = "";
     private _query: string = "";
     public set query(newQuery: string) {
-        let self = this;
+        this.previousQuery = this._query;
         this._query = newQuery;
-        if (this._query.trim() != "") {
-            // search
+        this.getStixResults(this._query);
+        this.getTechniqueResults(this._query);
+    }
 
-            //get master list of techniques and sub-techniques
-            let allTechniques = this.dataService.getDomain(this.viewModel.domainID).techniques;
-            for (let technique of allTechniques) {
-                allTechniques = allTechniques.concat(technique.subtechniques);
-            }
+    public get queryLength(): number {
+        return this._query.length;
+    }
+
+    public techniqueResults: Technique[] = [];
 
-            let results = allTechniques.filter(function(technique: Technique) {
+    constructor(private dataService: DataService) {
+        this.stixTypes = [];
+    }
+
+    ngOnInit() {
+        this.getStixData();
+        this.getTechniques();
+    }
+
+    // filterAndSort() takes an array of items and does the following:
+    // 1) if the query is empty, then it sorts the array
+    // 2) if the query is not empty, then it filters the already sorted array until nothing is left, or,
+    // the query is cleared out and empty again
+    filterAndSort(items: any[], query = "", sortAllTechniques = false) {
+        let self = this;
```

Comment:
It seems like deprecated items are not being filtered out of the sidebar results:
![image](https://user-images.githubusercontent.com/28868888/121416313-aba3fc80-c936-11eb-9dbe-4aa41fa1b6e3.png)
"bash_profile and .bashrc Mitigation" is a deprecated mitigation and shouldn't be shown.

Reply:
A quick way to tell (for mitigations specifically) is that if it ends with the word "mitigation" and the prefix looks like a technique name it's almost definitely a deprecated mitigation. There was an update which changed how mitigations behave and they used to follow that naming convention (`"[technique] Mitigation"`)